src/sql/src/plan/statement/ddl.rs

```diff
@@ -1482,6 +1410,108 @@ fn kafka_sink_builder(
+                let ccsr_config = kafka_util::generate_ccsr_client_config(
+                    schema_registry_url.clone(),
+                    config_options,
+                    ccsr_with_options,
+                )?;
+
+                Some((
+                    topic,
+                    KafkaSinkFormat::Avro {
+                        schema_registry_url,
+                        key_schema: None,
+                        value_schema: avro::get_debezium_transaction_schema().canonical_form(),
+                        ccsr_config,
+                    },
+                ))
+            }
+            None => {
+                // If a CONSISTENCY FORMAT is not provided, default to the FORMAT of the sink.
+                match sink_format {
+                    format @ KafkaSinkFormat::Avro { .. } => Some((topic, format.clone())),
+                    KafkaSinkFormat::Json => bail_unsupported!("CONSISTENCY FORMAT JSON"),
+                }
+            }
+            Some(other) => bail_unsupported!(format!("CONSISTENCY FORMAT {}", &other)),
+        },
+        None => {
+            // Support use of `consistency_topic` with option if the sink is Avro-formatted
+            // for backwards compatibility.
+            if reuse_topic | consistency_topic.is_some() {
+                match sink_format {
+                    KafkaSinkFormat::Avro {
+                        schema_registry_url,
+                        ccsr_config,
+                        ..
+                    } => {
+                        let consistency_topic = match consistency_topic {
+                            Some(topic) => topic,
+                            None => {
+                                let default_consistency_topic =
+                                    format!("{}-consistency", topic_prefix);
+                                debug!(
+                                    "Using default consistency topic '{}' for topic '{}'",
+                                    default_consistency_topic, topic_prefix
+                                );
+                                default_consistency_topic
+                            }
+                        };
+                        Some((
+                            consistency_topic,
+                            KafkaSinkFormat::Avro {
+                                schema_registry_url: schema_registry_url.clone(),
+                                key_schema: None,
+                                value_schema: avro::get_debezium_transaction_schema()
+                                    .canonical_form(),
+                                ccsr_config: ccsr_config.clone(),
+                            },
+                        ))
+                    }
+                    KafkaSinkFormat::Json => bail!("with FORMAT JSON, you need to manually specify a consistency topic using 'CONSISTENCY TOPIC consistency_topic CONSISTENCY FORMAT format'"),
+                }
+            } else {
```

Comment:
I would say "For FORMAT JSON" instead of "with FORMAT JSON", but it's a preference, not a hard rule.

Reply:
Your preference in these matters is definitely better than anything I do, changing... :sweat_smile: